smokey/src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -102,13 +113,32 @@ public ArmSubsystem() {
 
       // init smart motion and set positions if mag sensors are set
       this.initializeMotorsSmartMotion();
-      boolean isHorizontalArmAtOrBelowLowStop = (this.horizontalArmMageneticSensor.get() == false);
-      boolean isVerticalArmAtOrBelowLowStop = (this.verticalArmMageneticSensor.get() == false);
-      if(isHorizontalArmAtOrBelowLowStop){
-        this.horizontalEncoder.setPosition(0.0);
+
+      if(InstalledHardware.verticalArmBottomSensorInstalled) {
+        verticalArmBottomMageneticSensor = new DigitalInput(Constants.VirticalArmBottomMagneticSensor);
+        verticalArmBottomCorrectableEncoder = new CorrectableEncoderRevNeoPlusDigitalIoPort(
+          verticalEncoder,
+          verticalArmBottomMageneticSensor,
+          ArmSubsystem.convertVerticalArmExtensionFromMetersToTicks(ArmSubsystem.verticalArmBottomSensorPlacementAlongExtensionMeters),
+          true);
       }
-      if(isVerticalArmAtOrBelowLowStop){
-        this.verticalEncoder.setPosition(0.0);
+
+      if(InstalledHardware.verticalArmMiddleSensorInstalled) {
+        verticalArmMiddleMageneticSensor = new DigitalInput(Constants.VirticalArmMiddleMagneticSensor);
+        verticalArmMiddleCorrectableEncoder = new CorrectableEncoderRevNeoPlusDigitalIoPort(
+          verticalEncoder,
+          verticalArmMiddleMageneticSensor,
+          ArmSubsystem.convertVerticalArmExtensionFromMetersToTicks(ArmSubsystem.verticalArmMiddleSensorPlacementAlongExtensionMeters),
+          false);
```

Comment:
I don't understand why this parameter is false for the verticalArmMiddleSensor, but true for the other two sensors.

Reply:
So for the bottom angle/vertical arm sensor and the horizontal/extension arm sensor these two are at the end of the travel range.  So if the arm is in that position at the start, the assumption here is 'that is essentially zero'.  I'd be the first to admit that we might want to remove this assumption, but the thing is that the previous code makes this assumption as well ... so the intent here was to be effectively compatible with the old code.

The middle angle/vertical sensor, however, is different.  The tube could start down well below the sensor location and trigger the sensor, but a triggered sensor here does not imply the arm is at the 'middle position interface' ... so we can't 
 assume that we know where the arm location is (as much as we can about the sensors located at beginning or end of the allowable travel).

Reply:
got it.  would it be clearer to rename it: initalizeEncoderPositionWhileDioTriggered?  That way it is more clear that it happens the whole time the sensor is triggered, rather than on the change from not-triggered -> tirggered.